src/screens/DomainScreen/index.js

```diff
@@ -101,20 +101,21 @@ const DomainScreen = () => {
   };
 
   const getIFollow = async () => {
-    if (ifollow.length === 0) {
+    const isFollow = JSON.stringify(ifollow.map((i) => i.domain_id_followed)).includes(iddomain);
+    if (!isFollow) {
       const res = await getDomainIdIFollow();
       setIFollow(res.data, dispatch);
     } else {
-      setFollow(JSON.stringify(ifollow).includes(iddomain));
+      setFollow(isFollow);
     }
   };
 
   const init = async (withLoading = false) => {
     const domainName = dataDomain.og.domain;
+    if (withLoading) {
+      setLoading(true);
+    }
     if (domainName !== domainStore.selectedLastDomain) {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The loading state management logic has been moved inside the condition that checks if the current domain name is different from the last selected domain. This change makes sense because we only want to set the loading state when we're fetching new data for a different domain.



<!-- This is an auto-generated comment by OSS CodeRabbit -->